drizzle-schema/src/effect-schema.ts

```diff
@@ -0,0 +1,442 @@
+    infer Type,
+    infer Name,
+    infer TokenEncoded,
+    any,
+    infer HasDefault,
+    infer R
+  >
+  ? S.PropertySignature<
+    TokenType,
+    Type,
+    Name,
+    TokenEncoded,
+    ReplaceWith,
+    HasDefault,
+    R
+  >
+  : never;
+
+type CarryOverOptionality<From, To> = undefined extends From
+  ? null extends From
+  ? To | null | undefined
+  : To | undefined
+  : null extends From
+  ? To | null
+  : To;
+
+type PropertySignatureType<P> = P extends S.PropertySignature<
+  any,
+  infer Type,
+  any,
+  any,
+  any,
+  any,
+  any
+>
+  ? Type
+  : never;
+
+type BuildInsertSchema<
+  TTable extends Table,
+  TRefine extends InsertRefine<TTable> | {} = {},
+> = S.Schema<
+  Simplify<
+    S.Struct.Constructor<
+      {
+        [K in keyof InsertColumnPropertySignatures<TTable>]: InsertColumnPropertySignatures<TTable>[K];
+      } & {
+        [K in keyof TRefine &
+        string]: InsertColumnPropertySignatures<TTable>[K] extends S.PropertySignature<
+          any,
+          any,
+          any,
+          any,
+          any,
+          any
+        >
+        ? TRefine[K] extends S.Schema<any, any, any>
+        ? S.Schema<
+          CarryOverOptionality<
+            PropertySignatureType<
+              InsertColumnPropertySignatures<TTable>[K]
+            >,
+            S.Schema.Type<TRefine[K]>
+          >
+        >
+        : TRefine[K] extends (...a: any[]) => any
+        ? PropertySignatureReplaceType<
+          InsertColumnPropertySignatures<TTable>[K],
+          CarryOverOptionality<
+            PropertySignatureType<
+              InsertColumnPropertySignatures<TTable>[K]
+            >,
+            S.Schema.Type<ReturnType<TRefine[K]>>
+          >
+        >
+        : never
+        : TRefine[K];
+      }
+    >
+  >
+>;
+
+export function createInsertSchema<
+  TTable extends Table,
+  TRefine extends InsertRefine<TTable>,
+>(
+  table: TTable,
+  refine?: TRefine,
+): BuildInsertSchema<
+  TTable,
+  Equal<TRefine, InsertRefine<TTable>> extends true ? {} : TRefine
+> {
+  const columns = getTableColumns(table);
+  const columnEntries = Object.entries(columns);
+
+  let schemaEntries = Object.fromEntries(
+    columnEntries.map(([name, column]) => {
+      return [name, mapColumnToSchema(column)];
+    }),
+  );
+
+  if (refine) {
+    schemaEntries = Object.assign(
+      schemaEntries,
+      Object.fromEntries(
+        Object.entries(refine).map(([name, refineColumn]) => {
+          return [
+            name,
+            typeof refineColumn === 'function' &&
+              // guard against classes which are also functions, e.g. `Schema.Literal` class
+              !('annotations' in refineColumn)
+              ? refineColumn(schemaEntries as any)
+              : refineColumn,
+          ];
+        }),
+      ),
+    );
+  }
+
+  for (const [name, column] of columnEntries) {
+    if (!column.notNull) {
+      schemaEntries[name] = S.optional(S.NullOr(schemaEntries[name]!)) as any;
```

Comment:
```suggestion
      schemaEntries[name] = S.optional(schemaEntries[name]!, { nullable: true }) as any;
```

Reply:
is seems nullable doesn't add null to the type, I'll leave like it is for now

look at the "text" field
![image](https://github.com/user-attachments/assets/881f0b1b-2eca-421c-8041-6da57e6eff2b)
![image](https://github.com/user-attachments/assets/f4783520-ca9e-426a-b7c3-b4122dbd0227)

Reply:
Other way mate, the `Encoded` side, not the `Type` side lol